app/src/components/common/outcomes/index.tsx

```diff
@@ -71,9 +83,33 @@ const Outcomes = (props: Props) => {
     }
   }
 
+  const updateOutcomeName = (index: number, newName: string) => {
+    const newOutcome = {
+      ...outcomes[index],
+      name: newName,
+    }
+
+    const newOutcomes = [...outcomes.slice(0, index), newOutcome, ...outcomes.slice(index + 1)]
+    props.onChange(newOutcomes)
+  }
+
+  const messageErrorToRender = () => {
+    return (
+      <>
+        <ErrorStyled>{messageProbabilitiesError}</ErrorStyled>
+        <ErrorStyled>{messageEmptyNamesError}</ErrorStyled>
+      </>
+    )
+  }
+
   const outcomesToRender = props.outcomes.map((outcome: Outcome, index: number) => (
     <TwoColumnsRowExtraMargin key={index}>
-      <Textfield name={`outcome_${index}`} type="text" value={outcome.name} readOnly />
+      <Textfield
```

Comment:
We'll need to update the tests for this component.

Reply:
I added more tests